Ac 35 get metadata

Omermigdal opened the pull request:
1. removed Depends
2.Added getting metadata.
3. please comment on the TODO's and comment

Omermigdal committed:
- added get_metadata
- removed Depends() due to complications

Omermigdal committed:
Added TODO's

Omermigdal committed:
commented on the metadata return value

tal-alon commented on source/FilesServer/app/service.py:
```diff
@@ -22,10 +21,31 @@ def is_file_exists(
+async def s3_get_metadata(file_name: str) -> dict | None:
+    HTTP_headers = s3_get_head_object_HTTP_headers(file_name)
+    # TODO - check if only this is the content we want to return
+    return {
```
create a model to represent this data instead of a dictionary
you can create a models.py module

Omermigdal replied:
did it

tal-alon commented on source/FilesServer/app/service.py:
```diff
@@ -22,10 +21,31 @@ def is_file_exists(
+    ]["HTTPHeaders"]
+
+
+async def s3_get_metadata(file_name: str) -> dict | None:
```
this comment is not redundant but FYI, if you annotate a return value (or param for that matter) with list or dict or types like that, please add the specific type of list or dict, etc. 
e.g. list[int], dict[int, str] (integer keys and string values)

Omermigdal replied:
modified it to return FileMetadata

Omermigdal committed:
-  added FileMetadata model and changed code accordingly

Omermigdal committed:
HTTP->http

tal-alon commented on source/FilesServer/app/endpoints.py:
```diff
@@ -1,12 +1,16 @@
+# TODO - Check how to use Depends in order not to repeat get_settings() and get_client in each endpoint
 @router.get("/download", status_code=status.HTTP_200_OK)
-async def download(file_name: str) -> RedirectResponse:
+async def download(
```
reformat the code

Omermigdal replied:
i used ruff format . and ruff check . --fix and this is the outcome.. what am i doing wrong?

tal-alon replied:
ok, maybe i'm blind but this line looked a bit short to me. not long enough to be splatted

tal-alon commented on source/FilesServer/app/endpoints.py:
```diff
@@ -23,3 +27,14 @@ async def download(file_name: str) -> RedirectResponse:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST, detail="No file name provided"
+        )
+    return await service.s3_get_metadata(
```
this function might return a None, you need to handle this

Omermigdal replied:
done

tal-alon commented on source/FilesServer/app/service.py:
```diff
@@ -22,10 +22,25 @@ def is_file_exists(
+
+
+async def s3_get_metadata(file_name: str) -> FileMetadata | None:
+    http_headers = s3_get_head_object_http_headers(file_name)
```
same, you need to handle the optional None. maybe raising an exception will be easier. otherwise you need to check for None after every call

Omermigdal replied:
i think that i am handling this in the endpoint and also doesn't is_file _exists raises the needed error?

tal-alon commented on source/FilesServer/app/models.py:
```diff
@@ -0,0 +1,25 @@
+        return f"{size:.2f} KB"
+
+    @classmethod
+    def from_HTTP_headers(cls, file_name: str, HTTP_headers: dict) -> Self:
```
use lower case

tal-alon commented on source/FilesServer/app/service.py:
```diff
@@ -22,10 +22,25 @@ def is_file_exists(
-async def s3_get_presigned_url(
-    file_name: str | None, s3_client: boto3.client = Depends(get_s3_client)
-) -> str | None:
+def s3_get_head_object_http_headers(file_name: str) -> dict | None:
```
the function name is way too long and i'm not sure this function is even really necessary

tal-alon commented on source/FilesServer/app/service.py:
since every single function in this module need an instance of settings and s3 client it may be a good idea to restructure it an a class which will hold these as members

Omermigdal committed:
- created service claass
- added handling none in get file metadata
- reformatted code
- changed variables to lower case
-removed s3_get_head_object_http_headers
